Bio/motifs/__init__.py

```diff
@@ -440,14 +440,24 @@ def weblogo(self, fname, format="PNG", version="2.8.2", **kwds):
 
         """
         from Bio._py3k import urlopen, urlencode, Request
+        from Bio import Alphabet
+
+        if isinstance(self.alphabet, Alphabet.ProteinAlphabet):
```

Comment:
Can ``self.alphabet`` be a gapped alphabet? If so, use ``isinstance(Alphabet._get_base_alphabet(self.alphabet), Alphabet.ProteinAlphabet)`` here instead (and likewise for RNA and DNA).